File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {        
        Map<Character, Integer> map = new HashMap<>();
        for (int i = 0; i < 26; i++) {
            map.put((char) (i + 97), -1);
        }
    
        int[] answer = new int[s.length()];
        int idx = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (map.get(c) != -1) {
                answer[idx++] = i - map.get(c);
            } else {
                answer[idx++] = -1;
            }
            map.put(c, i);
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    alphabet = {chr(i + 97) : -1 for i in range(26)}
    for i in range(len(s)):
        if alphabet[s[i]] != -1:
            answer.append(i - alphabet[s[i]])
        else:
            answer.append(-1)
        alphabet[s[i]] = i 
    return answer